src/cmd/prepare.go

```diff
@@ -94,15 +95,43 @@ var prepareFindImages = &cobra.Command{
 	},
 }
 
+var prepareGenerateConfigFile = &cobra.Command{
+	Use:     "generate-config [FILENAME]",
+	Aliases: []string{"gc"},
+	Args:    cobra.MaximumNArgs(1),
+	Short:   "Generates a config file for Zarf",
+	Long: "Generates a Zarf config file for controlling how the Zarf CLI operates. Optionally accepts a filename to write the config to.\n\n" +
+		"The extension will determine the format of the config file, e.g. env-1.yaml, env-2.json, env-3.toml etc. \n" +
+		"Accepted extensions are json, toml, yaml.\n\n" +
+		"NOTE: This file must not already exist. If no filename is provided, the config will be written to the current working directory as zarf-config.toml.",
+	Run: func(cmd *cobra.Command, args []string) {
+		fileName := "zarf-config.toml"
+
+		// If a filename was provided, use that
+		if len(args) > 0 {
+			fileName = args[0]
+		}
+
+		if err := v.SafeWriteConfigAs(fileName); err != nil {
+			message.Fatalf(err, "Unable to write the config file %s, make sure the file doesn't already exist", fileName)
+		}
+	},
+}
+
 func init() {
+	initViper()
+
 	rootCmd.AddCommand(prepareCmd)
 	prepareCmd.AddCommand(prepareTransformGitLinks)
 	prepareCmd.AddCommand(prepareComputeFileSha256sum)
 	prepareCmd.AddCommand(prepareFindImages)
+	prepareCmd.AddCommand(prepareGenerateConfigFile)
+
+	v.SetDefault(V_PKG_CREATE_SET, map[string]string{})
 
 	prepareFindImages.Flags().StringVarP(&repoHelmChartPath, "repo-chart-path", "p", "", `If git repos hold helm charts, often found with gitops tools, specify the chart path, e.g. "/" or "/chart"`)
+	prepareFindImages.Flags().StringToStringVar(&config.CreateOptions.SetVariables, "set", v.GetStringMapString(V_PKG_CREATE_SET), "Specify package variables to set on the command line (KEY=value)")
 	prepareFindImages.Flags().StringVar(&config.CommonOptions.TempDirectory, "tmpdir", "", "Specify the temporary directory to use for intermediate files")
```

Comment:
Isn't this set at the root now?

Reply:
We actually have to load this like this (in create and prepare) due to how the functions initialize, I'll leave a comment

Reply:
I think it should also be in the output the user sees (i.e.)

```
Specify package variables to set on the command line as done for package create (KEY=value)
```

Reply:
ok... that seems like a bug. Also create doesn't have tmpdir?

Reply:
It's not a bug perse, it's just a unique quirk of how we are stringing viper -> cobra flags -> config variables. And it's always been that way, we just didn't notice before.

Reply:
Tmpdir was moved to a global command

![Screen Shot 2022-10-04 at 7 34 20 PM](https://user-images.githubusercontent.com/882485/193954802-f6831304-a695-4c7d-a0c7-59030abb0a1e.png)

Reply:
Removing that line doesn't change the docs...

```
// prepareFindImages.Flags().StringVar(&config.CommonOptions.TempDirectory, "tmpdir", "", "Specify the temporary directory to use for intermediate files")

```

Reply:
ah yes, fixed now--that was a product a bad merge conflict resolution 🤦